### ISSUE TITLE
reimplement convolutional (variational) autoencoders in flax for celeba

### ISSUE THREAD

**Issue by murphyk** [probml/pyprobml#1000]
figs 21.3 and 21.4 in book 2 (shown below) compare the reconstructon and unconditional sampling abilities of a convolutional AE and VAE. However these figures were generated by https://github.com/probml/pyprobml/blob/master/notebooks/book2/20/vae_compare_results.ipynb which relies on this somewhat complex pytorch lightning library: https://github.com/probml/pyprobml/tree/master/vae.

Please reproduce this basic demo using code that is similar to https://github.com/probml/pyprobml/blob/master/notebooks/book1/20/ae_mnist_conv_jax.ipynb.
To get the celeba data you will have to follow the instructions at https://github.com/probml/pyprobml/tree/master/vae#download-celeba-data.
For simplicity it is probably best to make a single colab that does everything, assuning that training can be done in under 5 minutes (if not, we may need to pretrain the models and store the checkpoints).

<img width="697" alt="Screen Shot 2022-07-12 at 2 48 13 PM" src="https://user-images.githubusercontent.com/4632336/178601411-f541881e-16bb-412a-bc30-2c3398c15fb3.png">

<img width="666" alt="Screen Shot 2022-07-12 at 2 48 18 PM" src="https://user-images.githubusercontent.com/4632336/178601431-6a4e3b8a-5b87-4e94-bdc9-1a71f3cf4973.png">

**Comment by murphyk** [comment]
It would also be helpful to make a separate notebook that compares AE and VAE on MNIST so we can reproduce figs 20.24 and 20.25 from book1 (see below).  You can factor out common code and put it in https://github.com/probml/probml-utils/.

Currently the reconstruction accuracy of the AE looks bad, which makes me suspect there is something wrong.
Also the current code relies on pytorch code in https://github.com/probml/probml-utils/blob/main/probml_utils/ae_mnist_conv.py and https://github.com/probml/probml-utils/blob/main/probml_utils/vae_conv_mnist.py, which I would like to deprecate.

<img width="680" alt="Screen Shot 2022-07-12 at 2 59 45 PM" src="https://user-images.githubusercontent.com/4632336/178602861-fbda193d-4040-4be4-a81b-00d7d62c4367.png">

**Comment by murphyk** [comment]
Also book 2 fig 20.7 uses the pytorch celebaA VAE code to do interpolation in latent space.

<img width="649" alt="Screen Shot 2022-07-12 at 3 35 22 PM" src="https://user-images.githubusercontent.com/4632336/178609885-0eae0cab-c833-4a7e-a5a8-917bdfeca280.png">

**Comment by murphyk** [comment]
Book 2 fig 20.8 will also need to be remade.
Currently uses https://github.com/probml/pyprobml/blob/master/notebooks/book2/20/vae_celebA_lightning.ipynb which uses the same pytorch code.
<img width="692" alt="Screen Shot 2022-07-12 at 3 36 47 PM" src="https://user-images.githubusercontent.com/4632336/178610091-f79aaa20-9ce8-4ebc-86bf-1d47e2110a20.png">

**Comment by nalzok** [comment]
It has been done

+ https://github.com/probml/pyprobml/blob/master/notebooks/book2/21/celeba_vae_ae_comparison.ipynb
+ https://github.com/probml/pyprobml/blob/master/notebooks/book2/21/mnist_vae_ae_comparison.ipynb
